src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+                {
+                    return ParseResult<StackFrameParameterDeclarationNode>.Abort;
+                }
+
+                typeIdentifier = new StackFrameArrayTypeExpression(typeIdentifier, arrayIdentifiers);
+            }
+
+            var identifier = _lexer.TryScanIdentifier(scanAtTrivia: false, scanLeadingWhitespace: true, scanTrailingWhitespace: true);
+            if (!identifier.HasValue)
+            {
+                return ParseResult<StackFrameParameterDeclarationNode>.Abort;
+            }
+
+            return new(new(typeIdentifier, identifier.Value));
+        }
+
+        /// <summary>
+        /// Parses the array rank specifiers for an identifier. 
+        /// Ex: string[,][]
+        ///           ^----^ both are array rank specifiers
+        ///                  0: "[,]
+        ///                  1: "[]"
+        /// </summary>
+        private ParseResult<ImmutableArray<StackFrameArrayRankSpecifier>> ParseArrayRankSpecifiers()
+        {
+            using var _ = ArrayBuilder<StackFrameArrayRankSpecifier>.GetInstance(out var builder);
+            using var _1 = ArrayBuilder<StackFrameToken>.GetInstance(out var commaBuilder);
+
+            while (true)
+            {
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.OpenBracketToken, scanTrailingWhitespace: true, out var openBracket))
+                {
+                    return new(builder.ToImmutable());
+                }
+
+                while (_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CommaToken, scanTrailingWhitespace: true, out var commaToken))
+                {
+                    commaBuilder.Add(commaToken);
+                }
+
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CloseBracketToken, scanTrailingWhitespace: true, out var closeBracket))
+                {
+                    return ParseResult<ImmutableArray<StackFrameArrayRankSpecifier>>.Abort;
+                }
+
+                builder.Add(new StackFrameArrayRankSpecifier(openBracket, closeBracket, commaBuilder.ToImmutableAndClear()));
+            }
+
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        /// <summary>
+        /// Parses text for a valid file path using valid file characters. It's very possible this includes a path that doesn't exist but
+        /// forms a valid path identifier. 
+        /// 
+        /// Can return if only a path is available but not line numbers, but throws if the value after the path is a colon as the expectation
+        /// is that line number should follow.
+        /// </summary>
+        private ParseResult<StackFrameFileInformationNode> TryParseFileInformation()
+        {
+            var path = _lexer.TryScanPath();
+            if (!path.HasValue)
+            {
+                return ParseResult<StackFrameFileInformationNode>.Empty;
+            }
+
+            if (path.Value.Kind != StackFrameKind.PathToken)
```

Comment:
hrmm... can this happen?  if so, doc when it does.  otherwise, if it shouldn't be possible, plz contract-throw.

Reply:
Changed this to 

```c#
if (path.Value.Kind == StackFrameKind.InvalidPathToken)
{
    return ParseResult<StackFrameFileInformationNode>.Abort;
}

Debug.Assert(path.Value.Kind == StackFrameKind.PathToken);
```